src/types.ts

```diff
@@ -213,3 +241,23 @@ export interface SelectRowEvent {
   checked: boolean;
   isShiftClick: boolean;
 }
+
+export type Dictionary<T> = Record<string, T>;
+
+export type GroupByDictionary<TRow> = Dictionary<{
+  childRows: readonly TRow[];
+  childGroups: readonly TRow[] | GroupByDictionary<TRow>;
+  startRowIndex: number;
+}>;
+
+export interface GroupRow<TRow> {
+  childRows: readonly TRow[];
+  id: string;
+  parentId: unknown;
+  groupKey: unknown;
```

Comment:
We should keep things consistent and either use `id` or `key` for those, not both.